src/protobuf-net.BuildTools/Internal/Utils.cs

```diff
@@ -41,25 +41,42 @@ internal static Location PickLocation(ref SyntaxNodeAnalysisContext context, ISy
         }
 
         internal const string ProtoBufNamespace = "ProtoBuf";
+        internal const string GenericCollectionsNamespace = "System.Collections.Generic";
+        internal const string ThreadingNamespace = "System.Threading";
+        internal const string ThreadingTasksNamespace = "System.Threading.Tasks";
 
-        internal static bool InProtoBufNamespace(this INamedTypeSymbol symbol)
+        internal static bool InGenericCollectionsNamespace(this ISymbol symbol)
+            => InNamespace(symbol, GenericCollectionsNamespace);
```

Comment:
shouldn't this be `InNamespace("System", "Collections", "Generic")` or similar? trying to test individual namespaces to avoid unnecessary concats